flytekit/annotated/task.py

```diff
@@ -86,7 +88,7 @@ def _compile(self, ctx: FlyteContext, *args, **kwargs):
         # TODO: Make the metadata name the full name of the (function)?
         sdk_node = Node(
             # TODO
-            id=f"node-{len(ctx.compilation_state.nodes)}",
+            id=f"{ctx.compilation_state.prefix}node-{len(ctx.compilation_state.nodes)}",
```

Comment:
should we just add a method to compilation context that generates the node id? and internally uses the prefix? This seems error prone

Reply:
i'm trying to resolve the case where you do

```
@workflow
def wf():
  @workflow
  def wf2():
    ...
```